### PR TITLE
Update all demos using the description from topic rather than the parameter

### PR DESCRIPTION
**Pull request by fmauch**
This will make sure things work after https://github.com/ros-controls/ros2_control/pull/1358 is merged.

I'll update this demo by demo whenever I find the time and check the demos according to their docs after modifying them.

### CODE REVIEW

File: example_14/hardware/rrbot_actuator_without_feedback.cpp [hunk 1]
```diff
@@ -183,21 +183,23 @@ hardware_interface::return_type RRBotActuatorWithoutFeedback::read(
...
-    rclcpp::get_logger("RRBotActuatorWithoutFeedback"), "Writing command: %f", hw_joint_command_);
-
-  // Simulate sending commands to the hardware
-  std::ostringstream data;
-  data << hw_joint_command_;
-  RCLCPP_INFO(
-    rclcpp::get_logger("RRBotActuatorWithoutFeedback"), "Sending data command: %s",
-    data.str().c_str());
-  send(sock_, data.str().c_str(), strlen(data.str().c_str()), 0);
+  if (!std::isnan(hw_joint_command_))
```
Comment (saikishor) [inline example_14/hardware/rrbot_actuator_without_feedback.cpp:186]:
```suggestion
  if (std::isfinite(hw_joint_command_))
```

May be `isfinite` is better here as it handles both `NaN` and `Infinity`

**Review by saikishor — CHANGES_REQUESTED** [review]

**Comment by fmauch** [comment]
Migrated all demos and manually tested them.

**Review by saikishor — APPROVED** [review]
The changes look good to me!

Let's merge once https://github.com/ros-controls/ros2_control/pull/1358 is merged!

**Comment by fmauch** [comment]
> The changes look good to me!
> 
> Let's merge once [ros-controls/ros2_control#1358](https://github.com/ros-controls/ros2_control/pull/1358) is merged!

I don't see any specific reason to wait for that, as this PR removes long-deprecated mechanisms and is in line with the docs again, but sure, it's not necessarily required before https://github.com/ros-controls/ros2_control/pull/1358 is merged.

**Comment by saikishor** [comment]
Sure, we can merge it. You have a point 👍🏽

**Review by christophfroehlich — APPROVED** [review]
LGTM, thank you for thinking of this! 
should we backport this? this should work even with humble?

**Comment by fmauch** [comment]
No, this will not work on humble because we subscribe to robot_description and not ~/robot_description on rolling.

**Pull request merged**